coc/troop.py

```diff
@@ -108,7 +108,7 @@ def __repr__(cls):
             ("id", cls.id),
         ]
         return "<%s %s>" % (
-            cls.__name__, " ".join("%s=%r" % t for t in attrs),)
+            cls.name, " ".join("%s=%r" % t for t in attrs),)
```

Comment:
No. __repr__ is supposed to output the class name. The troop name is handled by the join after that. The problem here is that __repr__ requires `self` as parameter and not `cls`. And than this becomes `self.__class__.__name__`. And we have the same problem in spell.py and hero.py.